src/statement.d

```diff
@@ -81,6 +81,19 @@ extern (C++) LabelStatement checkLabeledLoop(Scope* sc, Statement statement)
     return null;
 }
 
+Expression checkAssignmentAsCondition(Expression e)
+{
+    Expression e0;
+    Expression ec = Expression.extractLast(e, &e0);
+    if (ec.op == TOKassign)
+    {
+        e.error("assignment cannot be used as a condition, perhaps == was meant?");
+        auto ae = cast(AssignExp)ec;
```

Comment:
This is for more informative error report. Read the code comment I've added more.